src/metabase/api/api_key.clj

```diff
@@ -0,0 +1,75 @@
+(ns metabase.api.api-key
+  "/api/api-key endpoints for CRUD management of API Keys"
+  (:require
+   [compojure.core :refer [POST]]
+   [crypto.random :as crypto-random]
+   [metabase.api.common :as api]
+   [metabase.models.api-key :as api-key]
+   [metabase.models.permissions-group :as perms-group]
+   [metabase.models.user :as user]
+   [metabase.util :as u]
+   [metabase.util.honey-sql-2 :as h2x]
+   [metabase.util.malli.schema :as ms]
+   [toucan2.core :as t2]))
+
+
+(def ^:private user-first-name "API Key:")
+
+(defn- full-name [n]
+  (str user-first-name " " n))
```

Comment:
I don't really see why the comparison on full name is needed? The only thing we need to enforce is that API key names are unique.

Reply:
I wanted to avoid the case where someone creates an API key with the same last name as a user and is blocked from doing so. But moot now since the refactor!